src/lib/logger/sanitizer.ts

```diff
@@ -0,0 +1,21 @@
+import type { Sanitizer } from './types'
+
+export const sanitize = (input: string, sanitizer: Sanitizer) =>
+  sanitizer.secrets.reduce<string>((acc, secret) => {
+    if (typeof secret === 'string') {
+      acc = acc.replaceAll(secret, sanitizer.replacer)
+    }
+
+    return acc
+  }, input)
+
+export const stringify = (object: unknown, sanitizer: Sanitizer) => {
+  let message = ''
+
+  try {
+    const jsonString = JSON.stringify(object)
+    message = sanitize(jsonString, sanitizer)
```

Comment:
Is it ok we making exception and return empty string in case if obj is not serializable?

Reply:
I mean let's just throw exception in this case.
To prevent revealing secrets, let's just write smth common. Like: provided unserializable object

Reply:
I don't agree that it's a good idea to break the application flow because of incorrect logs. I've never seen a logger that can raise exceptions. Which is basically impossible since we have a strict type system and tests. Now it's done optimally: we just don't output logs that can't be output.